networkx/algorithms/tests/test_swap.py

```diff
@@ -12,13 +12,31 @@ def test_directed_edge_swap():
     assert out_degrees == sorted((n, d) for n, d in G.out_degree())
 
 
+def test_edge_cases_directed_edge_swap():
+    e = (
+        f"Maximum number of swap attempts \\(11\\) exceeded "
+        f"before desired swaps achieved \\(4\\)."
+    )
+    graph = nx.path_graph(4, create_using=nx.DiGraph)
+    with pytest.raises(nx.NetworkXAlgorithmError, match=e):
```

Comment:
Thanks @Mjh9122 - just one last note: no need to use f-strings here since there is no formatted insertion going on.